.github/workflows/msbuild.yml

```diff
@@ -55,11 +55,11 @@ jobs:
 
   integration-tests:
     needs: check-deep-tests
-    if: always() && ((needs.check-deep-tests.result == 'success' || contains(github.event.pull_request.labels.*.name, 'run-deep-tests')) || ( github.event_name == 'push' && ( github.ref_name == 'master' || vars.TEST_ON_FORK == 'true' )))
```

Comment:
Why is this change needed?

Reply:
To avoid running tests on the push to the master branch that happens after a merge of a PR that has just been tested as a PR event.